include/clblast_plugin.h

```diff
@@ -44,6 +44,47 @@ inline namespace version_1 {
 
 // =================================================================================================
 
+namespace database {
+
+using Parameters = std::unordered_map<std::string, size_t>;
+
+// Database tree leaf entry: describes a single device model
+struct Device {
+  // clGetDeviceInfo(CL_DEVICE_NAME) or "default"
+  std::string name;
+
+  // Actual parameters for this device and routine
+  Parameters parameters;
+};
+
+// Database tree vendor entry: group of entries for a single device vendor
+struct Vendor {
+  // clGetDeviceInfo(CL_DEVICE_TYPE) as string: "CPU", "GPU", "accelerator" or "default"
+  std::string type;
+
+  // clGetDeviceInfo(CL_DEVICE_VENDOR) or "default"
+  std::string name;
+
+  // Entries for this vendor
+  std::vector<Device> devices;
+};
+
+// Database tree routine entry: group of entries for a single routine tag
+struct Routine {
+  // Routine tag (see src/routines/*/*.cpp), multiple tags are used per each real BLAS routine
+  std::string kernel;
+
+  // Input data precision corresponding to these parameters
+  Precision precision;
+
+  // Entries for this routine
+  std::vector<Vendor> vendors;
+};
+
+} // namespace database
```

Comment:
All of the above is already elsewhere (`database.hpp`). I understand that it now becomes part of the interface, but perhaps the database code can use these new definitions instead? Problem would be the plugin namespace I guess. If not possible then it's no big deal I guess. Perhaps good to add a pointer to both files to the other file in case one of these structs ever gets changed.